pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -261,9 +289,26 @@ func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.S
 	if len(serverList) == 0 {
 		return nil, ErrNotFound
 	} else if len(serverList) > 1 {
-		return nil, ErrMultipleResults
+		var lastMatch *servers.Server
+		// try to find exactMatch
+		for _, server := range serverList {
+			if server.Name == exactMatch {
```

Comment:
But you're not using the node name, you're using os.Hostname(), which may or may not match the node name